client/src/pages/EpisodeShowPage/index.tsx

```diff
@@ -134,9 +140,22 @@ export class FormPagePresentational extends React.Component<any, any> {
 
   public constructor(props: any) {
     super(props);
+    this.state = {
+      timeLeft: Infinity,
+    };
   }
 
   public componentDidMount() {
+    if (this.props.match.params.timeLimit) {
+      if (!hasTimerStarted()) {
+        startTimer(Number(this.props.match.params.timeLimit));
+      }
+
+      setInterval(() => {
+        this.setState({timeLeft: howMuchTimeLeftOnTimer()});
+      }, 250);
```

Comment:
I suggest canceling this setInterval once the component is unmounted.
https://stackoverflow.com/questions/109086/stop-setinterval-call-in-javascript

Reply:
Another example: https://codepen.io/smonn/pen/KzezEw